vmc/sync_utils_test.go

```diff
@@ -0,0 +1,188 @@
+package vmc
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/vsphere-automation-sdk-go/lib/vapi/std/errors"
+	"github.com/vmware/vsphere-automation-sdk-go/services/vmc/model"
+	"testing"
+	"time"
+)
+
+func TestKeyedMutexLock(t *testing.T) {
+	var keyedMutex = KeyedMutex{}
+	var key1 = "key1"
+	var key2 = "key2"
+
+	lock1Obtained := false
+	lock2Obtained := false
+
+	var unlockFunction = keyedMutex.lock(key1)
+	// Try to flip the flag 1 in a separate thread
+	go func() {
+		keyedMutex.lock(key1)
+		lock1Obtained = true
+	}()
+	// Try to flip the flag 2 in a separate thread
+	go func() {
+		keyedMutex.lock(key2)
+		lock2Obtained = true
+	}()
+
+	// Give enough time for the separate threads to try to flip the flags
+	time.Sleep(500 * time.Millisecond)
+	assert.False(t, lock1Obtained)
+	assert.True(t, lock2Obtained)
+	// Test the unlock functionality
+	unlockFunction()
+	time.Sleep(500 * time.Millisecond)
+	assert.True(t, lock1Obtained)
+}
+
+type AuthenticatorMock struct {
```

Comment:
Those are not really mocks but stubs. 
You might want to generate the mocks actually to use proper mocks? See https://github.com/golang/mock

Reply:
Renamed to Stubs